### DISCUSSION TITLE
为什么需要指定速度？

### DISCUSSION THREAD

**Discussion by YOYOPole** [openvpi/SOME#4] [Q&A]
输入的速度是歌曲本身的bpm值？能否进行简化，谢谢。

**Comment by yqzhishen** [comment]
因为模型本身不包含识别BPM的功能，这也不是一个MIDI模型的职责范围，所以需要用户自行使用其他工具测量BPM后手动填入。实际落地后此类模型一般与编辑器、DAW等本身就含有谱面编辑功能的软件集成，因此可以使用工程中原本就有的曲速（甚至变速）。

BPM是多少并不影响MIDI文件中每个音符的实际位置和持续时间，仅影响它们在谱面上的表示。
